src/validation_workflow/docker/validation_docker.py

```diff
@@ -30,15 +27,19 @@ def __init__(self, args: ValidationArgs) -> None:
 
     def download_artifacts(self) -> bool:
 
-        # Check if the docker daemon is running; or else, bring it up
         if (self.is_container_daemon_running()):
 
             # STEP 1 . pull the images for OS and OSD
-            self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
-            self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
-
-            self.local_image_OS_id = self.get_image_id(self._OS_image_name, self.args.version)
-            self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, self.args.version)
+            if (self.args.using_staging_artifact_only):
+                self._OS_image_name = 'opensearchstaging/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchstaging/opensearch'
+                self._OSD_image_name = 'opensearchstaging/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchstaging/opensearch-dashboards'
+                self.local_image_OS_id = self.get_image_id(self._OS_image_name, ValidationArgs().stg_tag('opensearch').replace(" ", ""))
+                self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, ValidationArgs().stg_tag('opensearch_dashboards').replace(" ", ""))
+            else:
+                self._OS_image_name = 'opensearchproject/opensearch' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch'
+                self._OSD_image_name = 'opensearchproject/opensearch-dashboards' if self.args.docker_source == 'dockerhub' else 'public.ecr.aws/opensearchproject/opensearch-dashboards'
+                self.local_image_OS_id = self.get_image_id(self._OS_image_name, self.args.version)
+                self.local_image_OSD_id = self.get_image_id(self._OSD_image_name, self.args.version)
```

Comment:
I am seeing a lot of if else statements between dockerhub and ECR. Would recommend declaring class level variables for `self._OS_image_name` and `self._OSD_image_name`

The if else block can only be applied once

Reply:
I agree those looked ugly and not structured enough , let me work on optimizing them.. thanks @gaiksaya

Reply:
Hi, @gaiksaya, I am thinking it might be good to use a dictionary too, more clear on showing what is is trying to do I guess.. would that work from your end? Thank you?